src/app/components/ContentExcerpt.tsx

```diff
@@ -0,0 +1,37 @@
+import { Highlight } from '@openstax/highlighter/dist/api';
+import React from 'react';
+import styled from 'styled-components/macro';
+import addTargetBlankToLinks from '../content/utils/addTargetBlankToLinks';
+import { bodyCopyRegularStyle } from './Typography';
+
+// tslint:disable-next-line:variable-name
+export const HighlightContent = styled.div`
+  ${bodyCopyRegularStyle}
+  overflow: auto;
+
+  * {
+    overflow: initial;
+  }
+`;
+
+interface Props {
+  highlight: Highlight | string;
+  content: string;
+  className: string;
+}
+
+// tslint:disable-next-line:variable-name
+const ContentExcerpt = (props: Props) => {
```

Comment:
I was thinking, maybe we could make `ContentExcerpt` accept a render prop as children instead?
You could then make the signature like

```js
<ContentExcerpt content={highlight.highlightedContent}>
    {(parsedContent) => <HighlightContent dangerouslySetInnerHTML={parsedContent}/> }
</ContentExcerpt>
```

I think it could make the component focus on parsing content and not doing anything where-the-content-is-coming-from specific.

Reply:
what about the other props? So you mean like render two components?

Reply:
But does the `ContentExcerpt` care about other props? All it needs is html and data needed to parse that html, so book and page. All other props could be directly passed into the child.

Reply:
well data-highlight-id seams important, maybe className can always be the same but idk

Reply:
But you can pass the other props directly into the child like

```js
<ContentExcerpt content={highlight.highlightedContent}>
    {(parsedContent) => <HighlightContent dangerouslySetInnerHTML={parsedContent} data-highlight-id=""/> }
</ContentExcerpt>
```

ContentExcerpt wouldn't be aware of what you do with the parsed content.

Reply:
ok, um what about when the excerpt isn't a highlight?

Reply:
Well 

```js
<ContentExcerpt content={someHTML}>
    {(parsedContent) => <SomethingThatRendersParsedHTML otherProps={} dangerouslySetInnerHTML={parsedContent}/> }
</ContentExcerpt>
```

Reply:
@janKozie1 it seems like the purpose of your approach would be to pass an element that handles the display styling of the excerpt, where this component would only do the parsing.

i think your idea is reasonable, but i don't see why we couldn't accomplish the same thing at this point by simply extending `ContentExcerpt` as a styled component like:

const HighlightContent = styled(ContentExcerpt)`
  ...special highlight styles for the excerpt
`

which i feel is more straightforward and follows more common patterns.

if we ever run into a situation where this approach isn't working, we may consider coming back to your proposal here